### DISCUSSION TITLE
updating annotations.csv: good practices

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#298] [Q&A] [answered]
VanDam's 5-minute annotations had been imported with some of the start times mis-attributed (probably due to faulty metadata in homebank). A collaborator fixed the start times directly in annotations.csv. Is this "wrong", or is it okay to do it this way, rather than re-importing, since it won't change much (other than the importation date)?

**Comment by alecristia** [comment, marked as answer]
yes, because inside `converted/`, the csv's have onset and offset times that are based on that information. So you cannot just update annotations.csv

**Reply by lucasgautheron** [reply]
Yes, absolutely! It should never be updated manually.
I've just updated the docs to make it more obvious.

https://github.com/LAAC-LSCP/ChildProject/commit/e92b3be43fe7ce5c1a25a5b06610b9f6e2000ff4

**Reply by lucasgautheron** [reply]
Regarding vandam, the file that is used to determine the offsets during importation is the following: https://gin.g-node.org/EL1000/vandam/src/main/extra/0manifest2.csv

This file should be fixed, and then the importation script run again (`scripts/cha_import.py`)

**Reply by alecristia** [reply]
great! to fix this, I did:

```R
library(tidyverse)
library(readr)

raw <- read_csv("extra/0manifest2.csv")
start_times <- read_csv("extra/start_times.csv")
colnames(start_times)[2]<-"time_seek"

#fix typos
start_times$raw_filename=gsub("SC42/SC36_020806","SC42/SC42_020909",start_times$raw_filename)
start_times$raw_filename=gsub("KC10/KC","KC10/KC10",start_times$raw_filename)

raw[,-which(colnames(raw)=="time_seek")]->raw

merge(raw,start_times, all=T)->test
write.csv(test,"extra/0manifest2-corrected.csv")
```

Then edited `scripts/cha_import.py` to rely on "extra/0manifest2-corrected.csv", rather than "extra/0manifest2.csv"

```
child-project remove-annotations . --set cha/an1
```
But that failed:
> (ChildProjectVenv) (base) [acristia@oberon vandam]$ child-project remove-annotations . --set cha/an1
Traceback (most recent call last):
  File "/scratch1/home/acristia/ChildProjectVenv/bin/child-project", line 8, in <module>
    sys.exit(main())
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/ChildProject/cmdline.py", line 451, in main
    args.func(args)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/ChildProject/cmdline.py", line 290, in remove_annotations
    perform_validation(project, require_success=True, ignore_files=True)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/ChildProject/cmdline.py", line 44, in perform_validation
    errors, warnings = project.validate(**args)
TypeError: validate() got an unexpected keyword argument 'ignore_files'

So I tried directly to import, hoping that would overwrite the set:
```
datalad get extra/0manifest2-corrected.csv
datalad get annotations/cha/an1/raw/
python scripts/cha_import.py
child-project validate . --ignore-recordings
```

> warning: ./metadata/children.csv: unknown columns 'nae,ses,father_edu,twin,hearing_loss,original_ses' in children, exepected columns are: experiment,child_id,child_dob,location_id,child_sex,language,languages,mat_ed,fat_ed,car_ed,monoling,monoling_criterion,normative,normative_criterion,mother_id,father_id,order_of_birth,n_of_siblings,household_size,dob_criterion,dob_accuracy
warning: ./metadata/recordings.csv: unknown column 'cumulative_duration' in recordings, exepected columns are: experiment,child_id,date_iso,start_time,recording_device_type,recording_filename,duration,session_id,session_offset,recording_device_id,experimenter,location_id,its_filename,upl_filename,trs_filename,lena_id,might_feature_gaps,start_time_accuracy,noisy_setting,notes
validation successfully completed with 2 warning(s).

**Reply by lucasgautheron** [reply]
The issue with `child-project remove-annotations . --set cha/an1` is the same as that reported by Okko and has been fixed.

**Reply by lucasgautheron** [reply]
Apparently, it did not work for you because the .cha files were locally missing. Not sure why (I'd do `git restore annotations/cha/an1/raw` and `datalad get annotations/cha/an1/raw` to sort this out)

I have run the importation script myself and pushed the changes anyway (worked fine)
